fault-tolerance/src/main/java/io/helidon/faulttolerance/FaultTolerance.java

```diff
@@ -145,6 +147,32 @@ static Throwable cause(Throwable throwable) {
         return throwable;
     }
 
+    /**
+     * Establish a dependency between a source (stage) and a dependent (future). The
+     * dependent shall complete (normally or exceptionally) based on the source stage.
+     * The source stage shall be cancelled if the dependent is cancelled. The {@code
+     * mayInterruptIfRunning} flag is always set to {@code true} during cancellation.
+     *
+     * @param source the source stage
+     * @param dependent the dependent future
+     * @param <T> type of result
+     */
+    static <T> CompletableFuture<T> createDependency(CompletionStage<T> source, CompletableFuture<T> dependent) {
+        source.whenComplete((o, t) -> {
+            if (t != null) {
+                dependent.completeExceptionally(t);
+            } else {
+                dependent.complete(o);
+            }
+        });
+        dependent.whenComplete((o, t) -> {
+            if (t instanceof CancellationException) {
+                source.toCompletableFuture().cancel(true);
+            }
+        });
```

Comment:
Maybe using `CompletableFuture#isCancelled` would be safer, it's quite improbable but what if someone does:
```java
CompletionStage<Void> source = new CompletableFuture<>();
CompletableFuture<Void> dependant = new CompletableFuture<>();
createDependency(source, dependant);
dependent.completeExceptionally(new CancellationException());
```